cli/emit.rs

```diff
@@ -276,11 +289,80 @@ impl<'a> EmitParsedSourceHelper<'a> {
       }
     };
     debug_assert!(transpiled_source.source_map.is_none());
-    self.0.emit_cache.set_emit_code(
-      specifier,
-      source_hash,
-      &transpiled_source.source,
-    );
+    if should_cache {
+      self.0.emit_cache.set_emit_code(
+        specifier,
+        source_hash,
+        &transpiled_source.source,
+      );
+    }
     transpiled_source.source.into_boxed_slice().into()
   }
 }
+
+fn has_import_assertion(parsed_source: &deno_ast::ParsedSource) -> bool {
+  fn has_import_assertion(text: &str) -> bool {
+    // good enough
+    text.contains(" assert ") && !text.contains(" with ")
+  }
+
+  fn warn_import_attribute(
+    parsed_source: &deno_ast::ParsedSource,
+    range: SourceRange,
+  ) {
+    let text_info = parsed_source.text_info_lazy();
+    let loc = text_info.line_and_column_display(range.start);
+    deno_runtime::import_assertion_callback(
+      deno_core::ImportAssertionsSupportCustomCallbackArgs {
+        maybe_specifier: Some(parsed_source.specifier().to_string()),
+        maybe_line_number: Some(loc.line_number),
+        column_number: loc.column_number,
+        maybe_source_line: Some(range.text_fast(text_info).to_string()),
+      },
+    )
+  }
+
+  let Some(module) = parsed_source.program_ref().as_module() else {
+    return false;
+  };
+
+  let mut had_import_assertion = false;
+  for item in &module.body {
+    match item {
+      deno_ast::swc::ast::ModuleItem::ModuleDecl(decl) => match decl {
+        deno_ast::swc::ast::ModuleDecl::Import(n) => {
+          if n.with.is_some()
```

Comment:
Can you also handle `import("./data.json", { assert: { type: "json" }})`?

Reply:
Oh, that's annoying. I'll update it.

Reply:
We don't need to do it for that because v8 will warn about it.